jep/0000/README.adoc

```diff
@@ -0,0 +1,316 @@
+curl -H "Authorization: Bearer $TOKEN" -X DELETE https://api.github.com/repos/$repo/git/refs/tags/$tag
+----
+
+Now knowing that it has been called from an App with write permissions,
+such as the deployment Action,
+it can create a new Artifactory token with a short expiry (say one hour)
+granted permission only to upload to the paths defined for this repository in RPU
+and return that token in its response.
+The Action would then bind this token to an environment variable for use from `settings.xml`.
+
+== Backwards Compatibility
+
+The link:https://github.com/jenkins-infra/update-center2[Jenkins update center generator] requires no modifications:
+releases deployed by this JEP’s mechanism appear in the regular Artifactory `releases` repository,
+using unusual but perfectly legal release version numbers.
+
+The Jenkins plugin manager should require no modifications since it will be merely presented with valid-looking releases from the update center generator.
+The mechanism by which those releases were built and deployed is irrelevant.
+
+The link:https://github.com/jenkins-infra/pipeline-library[stock Pipeline library] can be used as is,
+or with arbitrary modifications:
+customizations to how tests are run and so on would affect whether and how quickly ci.jenkins.io produces a passing commit status,
+without any interaction with the subsequent deployment.
+(The library already tolerates incremental versions from JEP-305;
+`infra.maybeDeployIncrementals` could be amended to skip deployment from `master` when `changelist.format` is defined,
+to avoid redundantly deploying the same bits to `incrementals` as would anyway be deployed to `releases`.)
+
+The link:https://github.com/jenkinsci/plugin-compat-tester[Plugin Compatibility Tester (PCT)] should require no modifications
+to test plugins deployed by this JEP’s mechanism, or plugins depending on such releases:
+it has long since been fixed to tolerate incremental versions and JEP-305’s use of `flatten-maven-plugin`.
+
+== Security
+
+The GitHub runner is solely responsible for rebuilding binary artifacts (such as plugin `*.hpi`) from sources.
+This defends against certain supply-chain attacks:
+if ci.jenkins.io were compromised, at worst this could result in components with test failures being deployed.
+The deployed binaries would still have been built from the source files stored on GitHub.
+
+Currently we presume that component maintainers are not maliciously inserting backdoors into manually deployed binaries.
+So long as maintainers are granted direct access to Artifactory as well as the option to use CD, trusting them is unavoidable,
+and it is desirable to offer this option to maintainers in order for example to produce backport releases
+(since the current system only handles `master` pushes).
+If a maintainer were _not_ given Artifactory credentials (or simply not listed in RPU),
```

Comment:
I am not sure I follow. What sort of “decisions about the component”? If using MRP, you must have both a valid Artifactory token _and_ write permission to the repository. (Maybe some malicious uses of a saved secret require owner permission rather than merely write?)

Reply:
> What sort of “decisions about the component”?

Who else gets to be a maintainer. Whether to suspend distribution. How to configure the repo if they're not a maintainer. Who to invite as a committer.

As the YAML files with upload permissions are the best reference we have had for a long time for who gets such decisions for plugins, their meaning quickly grew beyond the simple, technical "allowed to upload".

As presumably there will always be someone who will have to make these decisions for a component unless it is abandoned, there will always be users listed in YAML files, even if all releases are done via GitHub.

I've discussed with @oleg-nenashev previously that we need to separate the technical ability to upload new releases from the "ownership" of a component (and perhaps define the latter better), but I so far it looks like neither of us found the time to do this.

Reply:
OK, I think I follow what you are saying now, and will try to update the document to reflect that.